cmd/kubeadm/app/cmd/init.go

```diff
@@ -121,6 +124,8 @@ type initData struct {
 	client                clientset.Interface
 	waiter                apiclient.Waiter
 	outputWriter          io.Writer
+	certificates          map[string]*x509.Certificate
+	privateKeys           map[string]*rsa.PrivateKey
```

Comment:
Wondering which is worse from security stand point. Is it storing single copies of private keys with a longer life than usual, or having several short lived copies, that are not securely wiped out upon garbage collection.

Reply:
Maybe we can cleanup the certs in the end of phase.

Reply:
Or move the cache for the cert phase scope.

Reply:
it doesn't matter that much because we are reading from disk as the source of truth (e.g. an attacker might also be able to read from disk), but in terms of the security of the process memory the shorter lived copies are better, because the go GC is pretty good, while the alternative gives a much bigger time window for an attack as now we store the certs in a map.

i would leave this as is for now without further cleanups.